### PR TITLE
[feature] detach from ASG

### PR DESCRIPTION
**Pull request by egeland**
The idea for this came from @Sudiksha and was based on their initial PR.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -92,6 +101,13 @@ if [ "${SEMATEXT_URL}" != "" ]; then
   curl -X POST --data "{\"message\":\"$MESSAGE\",\"title\":\"Spot Termination ${CLUSTER_INFO}\",\"host\":\"${NODE_NAME}\",\"Instance\":\"${INSTANCE_ID}\", \"Availability Zone\":\"${AZ}\", \"type\":\"aws_spot_instance_terminated\"}" ${SEMATEXT_URL}
 fi
 
+# Detach from autoscaling group, which will cause faster replacement
+# We do this in parallel with the drain (see the & at the end of the command).
+if [ "${DETACH_ASG}" != "false" ]; then
```
[the remark this answers is not in the document]
Comment (egeland) [reply entrypoint.sh]: Will the `ASG_NAME=$(aws --output text --region ${REGION} autoscaling describe-auto-scaling-instances --instance-ids ${INSTANCE_ID} |awk '{print $2}')` return the empty string in spot fleet?
If not, maybe I should add a note to not use this feature with spot fleet?

[the remark this answers is not in the document]
Comment (mumoshu) [reply entrypoint.sh]: Yes, it should. The behavior should be same you've provided an inexistent instance id like:

```
aws autoscaling describe-auto-scaling-instances --instance-ids foo --output text
```

Also, spot fleet managed instances won't have a corresponding auto-scaling-instance.

File: README.md [hunk 2]
```diff
@@ -113,8 +113,18 @@ Example Pod Spec:
             value: "https://event-receiver.sematext.com/APPLICATION_TOKEN/event"
           - name: CLUSTER
             value: development
+          - name: DETACH_ASG
+            value: "yes"
```
Comment (Sudiksha) [inline README.md]: Can we replace "yes" with "true". (As default value is "false")

**Comment by egeland** [comment]
@mumoshu , would you mind approving?

Also, could you give @matthope access, so he can help us out?

**Comment by mumoshu** [comment]
Thanks for your efforts! LGTM.

@egeland Sure. Just invied @matthope to as a collaborator of this repository 😄

**Comment by Sudiksha** [comment]
@egeland @mumoshu I see you have pushed a new image `kubeaws/kube-spot-termination-notice-handler:latest` with these changes. Thank you.
Can you please push these changes as a tag version rather than latest? Its better from reference perspective.

**Comment by Sudiksha** [comment]
@egeland Can you please help in adding the new image with a verion tag?

**Pull request merged**